docs/index.md

```diff
@@ -3,85 +3,326 @@ hide:
+@me.page()
+def object_detector():
+    state = me.state(State)
 
-__Step 3:__ Run the app
+    me.text("Real-time Object Detection", type="headline-4")
+    me.uploader(label="Upload an image", on_upload=on_image_upload)
 
-```sh
-mesop main.py
+    if state.image_data:
+        me.image(src=f"data:image/jpeg;base64,{state.image_data}")
+
+    if state.detections:
+        me.text("Detected Objects:", type="headline-5")
+        for detection in state.detections:
+            detection_component(detection)
+
+def detection_component(detection):
+    me.text(f"{detection.obj}: {detection.confidence:.2f}")
+
+def on_image_upload(e: me.UploadEvent):
+    state = me.state(State)
+    state.image_data = base64.b64encode(e.file.read()).decode()
+    state.detections = detect_objects(e.file)
 ```
 
-## Next Steps
+<section class="container pink-gradient-background">
+  <div class="section-chip pink">Building blocks</div>
+  <div class="section-headline">
+    Ready-to-use components
+  </div>
+  <div class="section-body">
+    Mesop provides a versatile range of <a href="./components/">30 components</a>, from low-level building blocks to high-level, AI-focused components. This flexibility lets you rapidly prototype ML apps or build custom UIs, all within a single framework that adapts to your project's use case.
+
+    <a class="section-button" href="./components/chat">
+      AI components
+      <div class="section-button-subtext">e.g. chat, text to image</div>
+    </a>
+    <a class="section-button" href="./components/input">
+      Form components
+      <div class="section-button-subtext">e,g. input, checkbox, radio</div>
+    </a>
+    <a class="section-button" href="./components/table">
+      Data display components
+      <div class="section-button-subtext">e,g. table, plot</div>
+    </a>
+  </div>
+</section>
+
+<section class="container teal-gradient-background">
+  <div class="section-chip teal">Build anything</div>
+  <div class="section-headline">
+    Build any user interface you can imagine
+  </div>
+  <div class="section-body">
+With Mesop, you can build virtually any web-based user interface or application you can imagine. From quick prototypes to enterprise tools, Mesop provides the customizability to bring your ideas to life.
+  </div>
+  <div class="center">
+  <!-- UPDATE: -->
+  <iframe src="http://localhost:32123/embed/chat_inputs" class="chat-inputs-iframe box-shadow"></iframe>
+  </div>
+</section>
+
+<section class="container amber-gradient-background">
+  <div class="section-chip amber">Extensible</div>
+  <div class="section-headline">
+  Seamlessly integrate JS with web components
+  </div>
+  <div class="section-body">
+With Mesop <a href="./web_components/">web components</a>, you get the best of both worlds: the simplicity and power of Python, with the ability to drop down into JavaScript when you need that extra control.  Use existing JS libraries - all while keeping your core logic in Python.
```

Comment:
The second sentence feels a bit disjoint. Second second sentence feels like tag line that didn't chosen as a tag line. But was still important to keep in.